server/app.js

```diff
@@ -9,12 +9,12 @@ const app = express();
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(bodyParser.json());
 app.set('port', process.env.PORT || 4001);
+app.use(cookieParser());
+app.use(controllers);
 app.use(express.static(path.join(__dirname, '..', 'client', 'build')));
 
-app.get('/', (req, res) => {
+app.get('*', (req, res) => {
```

Comment:
@amusameh Why did you use `*` instead of `/`?

Reply:
Whenever I type any route in the address bar(away from `'/' `) it will give `404`, so we had to use `*` to match all the routes 
https://stackoverflow.com/questions/35763562/allowing-expess-to-pass-routes-to-react-router

Do you have another way to do it? What do you think?

@NoureldeanSaed  @tomduggan85

Reply:
@amusameh I guess you could just remove the whole string `'*'` and the page will be rendered at all requests.
But still, this will cause you a problem. That you can no longer make any other API calls to your server except for this one (The one that handles sending `index.html`). For all other routes you'll get an error stating that `you cannot set headers after they are sent to the client`.

Reply:
I’m not as familiar with express but theoretically this seems like it would work - since your controllers are now above this app.get call, your API calls will already be caught and they’d never reach this point, I would think. 
Have you tried this out yet? Does it look like it’s working?
If it’s not working, my only other idea is to write out all of the client routes that you need to support here, as individual app.get calls (or maybe app.get lets you pass an array of route strings?) but that would be a little ugly and hard to maintain.

Reply:
@tomduggan85 Yeah that skipped my mind, the fact that the controllers are above this generic controller does solve the problem I mentioned earlier.

Reply:
A comment might help, since you can’t see the other routes here and how they’re intended to come before the app.get.

Reply:
@tomduggan85 
Yeah it actually works without any problems(at least so far :smile:). Noureldean question made me wonder if there is a better way to do it so I asked.
I'll be adding a comment to make it clear 
thanks